### PR TITLE
GUI: Improve how shaders are handled

### PR DESCRIPTION
**Pull request by ccawley2011**
This has not been tested on PS Vita, and has only been tested on desktop with a modified version of SurfaceSdlGraphicsManager. This also doesn't account for users with old config files.

### CODE REVIEW

**Comment by rsn8887** [comment]
Doesn't compile with --host=psp2, I get:
```
../scummvm/backends/graphics/psp2sdl/psp2sdl-graphics.cpp: In constructor 'PSP2SdlGraphicsManager::PSP2SdlGraphicsManager(SdlEventSource*, SdlWindow*)':
../scummvm/backends/graphics/psp2sdl/psp2sdl-graphics.cpp:81:2: error: class 'PSP2SdlGraphicsManager' does not have any field named '_currentShader'
  _currentShader(GFX_SHADER_NONE) {
```

File: backends/graphics/psp2sdl/psp2sdl-graphics.cpp [hunk 1]
```diff
@@ -77,7 +77,8 @@ PSP2SdlGraphicsManager::PSP2SdlGraphicsManager(SdlEventSource *sdlEventSource, S
 	:
 	SurfaceSdlGraphicsManager(sdlEventSource, window),
 	_vitatex_hwscreen(nullptr),
-	_sdlpixels_hwscreen(nullptr) {
+	_sdlpixels_hwscreen(nullptr),
+	_currentShader(GFX_SHADER_NONE) {
```
Comment (rsn8887) [inline backends/graphics/psp2sdl/psp2sdl-graphics.cpp]: This should be removed, because _currentShader is a member of the parent.

**Comment by ccawley2011** [comment]
The compilation issues have been fixed. I also extended this PR to move the shader code out of `SurfaceSdlGraphicsManager`, since it shouldn't be needed any more.

**Comment by rsn8887** [comment]
Tested working on Vita, ready to merge!

**Pull request merged**